py/server/deephaven/stream/table_publisher.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+"""The table_publisher module supports publishing Deephaven Tables into blink Tables."""
+
+import jpy
+import traceback
+
+from typing import Callable, Dict, Optional
+
+from deephaven._wrapper import JObjectWrapper
+from deephaven.column import Column
+from deephaven.dtypes import DType
+from deephaven.execution_context import get_exec_ctx
+from deephaven.jcompat import j_runnable
+from deephaven.table import Table
+from deephaven.update_graph import UpdateGraph
+
+_JTableDefinition = jpy.get_type("io.deephaven.engine.table.TableDefinition")
+_JTablePublisher = jpy.get_type("io.deephaven.stream.TablePublisher")
+_JRuntimeException = jpy.get_type("java.lang.RuntimeException")
+
+
+class TablePublisher(JObjectWrapper):
+    """Supports publishing Table data."""
+
+    j_object_type = _JTablePublisher
+
+    def __init__(self, j_table_publisher: jpy.JType):
+        self.j_table_publisher = j_table_publisher
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_table_publisher
+
+    @property
+    def is_alive(self) -> bool:
+        """Checks whether this publisher is alive; if False, the publisher should stop publishing new data and release
+        any related resources as soon as practicable since publishing won't have any downstream effects.
+
+        Once this is False, it will always remain False. For more prompt notifications, publishers may
+        prefer to use on_shutdown_callback during construction.
+
+        Returns:
+            if this publisher is alive
+        """
+        return self.j_table_publisher.isAlive()
+
+    def add(self, table: Table):
+        """Adds a snapshot of the data from table into the blink table according to the blink table's definition.
+
+        All of the data from table will be:
+
+        1. consistent with a point in time
+        2. fully contained in a single blink table's update cycle
+        3. non-interleaved with any other calls to add (concurrent, or not)
+
+        Args:
+            table (Table): the table to add
+        """
+        self.j_table_publisher.add(table.j_table)
+
+    def publish_failure(self, failure: Exception):
+        """Publish a failure for notification to the listeners of the blink table. Future calls to add will silently
+        return. Will cause the on-shutdown callback to be invoked if it hasn't already been invoked.
```

Comment:
I don't think this will be clear to a user.  How about:
```suggestion
        """Indicate that data publication has failed.  Blink table listeners will be notified of the failure, the on-shutdown callback will be invoked if it hasn't already been, the listener will no longer be alive, and future calls to add will silently return without publishing.
```